tools/dissector/sample_dissector.cpp

```diff
@@ -91,261 +250,523 @@ namespace OpenDDS
+        p.serializer >> short_value;
+        s << short_value;
+        break;
+
+      case Long:
+        ACE_CDR::Long long_value;
+        p.serializer >> long_value;
+        s << long_value;
+        break;
+
+      case LongLong:
+        ACE_CDR::LongLong longlong_value;
+        p.serializer >> longlong_value;
+        s << longlong_value;
+        break;
+
+      case UShort:
+        ACE_CDR::UShort ushort_value;
+        p.serializer >> ushort_value;
+        s << ushort_value;
+        break;
+
+      case ULong:
+        ACE_CDR::ULong ulong_value;
+        p.serializer >> ulong_value;
+        s << ulong_value;
+        break;
+
+      case ULongLong:
+        ACE_CDR::ULongLong ulonglong_value;
+        p.serializer >> ulonglong_value;
+        s << ulonglong_value;
+        break;
+
+      case Float:
+        ACE_CDR::Float float_value;
+        p.serializer >> float_value;
+        s << float_value;
+        break;
+
+      case Double:
+        ACE_CDR::Double double_value;
+        p.serializer >> double_value;
+        s << double_value;
+        break;
+
+      case LongDouble:
+        ACE_CDR::LongDouble longdouble_value;
+        p.serializer >> longdouble_value;
+        s << longdouble_value;
+        break;
+
+      case String:
+        ACE_CDR::Char * string_value;
+        string_value = NULL;
+        p.serializer.read_string(string_value);
+        s << string_value;
+        delete [] string_value;
+        break;
+
+      case WString:
+        size_t wstring_length;
+        ACE_CDR::WChar * wstring_value;
+        wstring_value = NULL;
+        wstring_length = p.serializer.read_string(wstring_value);
+        utf16_to_utf8(utf8, wstring_value, wstring_length);
+        delete [] wstring_value;
+        s << utf8;
+        break;
+
+      case Enumeration:
+        break; // only the label is used, not directly presented
+
+      case Undefined:
+        s << "type undefined";
+
+      }
+
+      p.offset += (p.buffer_pos() - location);
+    }
+
+#define ADD_FIELD_PARAMS params.tree, hf, params.tvb, params.offset, (gint) len
     size_t
-    Sample_Field::compute_field_length (guint8 *data)
-    {
+    Sample_Field::dissect_i (Wireshark_Bundle &params, bool recur) {
+
       size_t len = 0;
-      switch (this->type_id_)
-        {
-        case Char:
-          {
-            len = sizeof (ACE_CDR::Char);
-            break;
-          }
-        case Boolean:
-          {
-            len = sizeof(ACE_CDR::Boolean);
+
+      if (!this->nested_) {
+
+        int hf = -1;
+        if (!params.get_size_only) {
+           hf = get_hf();
+        }
+
+        if (hf == -1 && !params.get_size_only) {
+          throw Sample_Dissector_Error(
+            get_ns()  + " is not a registered wireshark field."
+          );
+        } else {
+
+          size_t location = params.buffer_pos();
+
+          // Set Field
+          switch (this->type_id_) {
+
+          case Sample_Field::Boolean:
```

Comment:
This code is in the class Sample_Field, so the Sample_Field:: scope shouldn't be needed here.